pandas/src/algos_rank_helper.pxi.in

```diff
@@ -0,0 +1,369 @@
+"""
+Template for each `dtype` helper function for rank
+
+WARNING: DO NOT edit .pxi FILE directly, .pxi is generated from .pxi.in
+"""
+
+#----------------------------------------------------------------------
+# rank_1d, rank_2d
+#----------------------------------------------------------------------
+
+{{py:
+
+# name ctype pos_nan_value neg_nan_value
+dtypes = [('generic', 'object', 'Infinity()', 'NegInfinity()'),
+          ('float64', 'float64_t', 'np.inf', '-np.inf'),
+          ('uint64', 'uint64_t', '', ''),
+          ('int64', 'int64_t', 'np.iinfo(np.int64).max',
+           'np.iinfo(np.int64).min')]
+
+}}
+
+{{for name, ctype, pos_nan_value, neg_nan_value in dtypes}}
+
+
+{{if name == 'generic'}}
+
+
+def rank_1d_{{name}}(object in_arr, bint retry=1, ties_method='average',
+                     ascending=True, na_option='keep', pct=False):
+{{else}}
+
```

Comment:
prob could just have the same signature for object (and just ignore that arg no?)